test/unittests/Optimizer/TestOptimizerOverrides.cpp

```diff
@@ -2,13 +2,128 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include "ttmlir/Dialect/TTNN/Utils/OptimizerOverrides.h"
+#include "llvm/Support/CommandLine.h"
 #include <gtest/gtest.h>
 
-#include "ttmlir/Dialect/TTNN/Utils/OptimizerOverrides.h"
+#include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"
+#include "ttmlir/Dialect/TTNN/IR/TTNNOpsAttrs.h"
 
 using namespace mlir::tt::ttnn;
 
-class TestOptimizerOverrides : public ::testing::Test {
+class OutputLayoutOverrideTest : public ::testing::Test {
+protected:
+  llvm::cl::opt<std::string> OverrideOutputLayoutOption{
+      "override-output-layout"};
+  OutputLayoutOverrideParser parser{OverrideOutputLayoutOption};
+  llvm::StringMap<OutputLayoutOverrideParams> parsedOverride;
```

Comment:
@odjuricicTT, I think this might have broken macOS build: https://github.com/tenstorrent/tt-mlir/actions/runs/12269439074/job/34232997171

```
OptimizerTests: CommandLine Error: Option 'override-output-layout' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
```

Perhaps this fixture gets instantiated many times?

Reply:
Seems so. Thanks for pointing it out. Will take a look asap.

Reply:
I tried a potential fix here: https://github.com/tenstorrent/tt-mlir/pull/1589

But I'm unable to repro locally, it feels like a build config issue actually, I think the usage of `parser.parse` is correct.

Reply:
It seems to be some strange behaviour where if two specific tests exist, one of them fails. Once one is deleted, all tests pass. Also unable to repro locally. Will delete one of the tests for now to unblock further work, and take a deeper look later.

Temporary fix: https://github.com/tenstorrent/tt-mlir/pull/1598

Issue for future: https://github.com/tenstorrent/tt-mlir/issues/1597